screenapiexample/src/androidTest/java/net/gini/android/vision/screen/screenshots/ScreenshotsTest.java

```diff
@@ -0,0 +1,129 @@
+package net.gini.android.vision.screen.screenshots;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import static net.gini.android.vision.screen.screenshots.Helper.openApp;
+import static net.gini.android.vision.screen.screenshots.ScreenshotHelper.screenshotFileForBitBar;
+import static net.gini.android.vision.screen.screenshots.ScreenshotHelper.takeUIAutomatorScreenshot;
+import static net.gini.android.vision.screen.testhelper.PermissionsHelper.grantCameraPermission;
+
+import android.os.RemoteException;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.filters.SdkSuppress;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiDevice;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+
+import net.gini.android.vision.screen.screenshots.screens.screenapi.AnalysisScreen;
+import net.gini.android.vision.screen.screenshots.screens.screenapi.CameraScreen;
+import net.gini.android.vision.screen.screenshots.screens.screenapi.MainScreen;
+import net.gini.android.vision.screen.screenshots.screens.screenapi.OnboardingScreen;
+import net.gini.android.vision.screen.screenshots.screens.screenapi.ReviewScreen;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.util.Date;
+
+@SdkSuppress(minSdkVersion = 18)
+@RunWith(AndroidJUnit4.class)
+public class ScreenshotsTest {
+
+    private static final String SCREEN_API_EXAMPLE_APP = "net.gini.android.vision.screenapiexample";
+
+    private UiDevice mDevice;
+
+    @Before
+    public void setUp() throws Exception {
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+        assertThat(mDevice).isNotNull();
+        // Start from the home screen
```

Comment:
Just a doubt. Why is it needed to start from the home screen?

Reply:
The testing framework is not starting the application automatically, but the gradle tasks executing the tests installs the app before test execution. In order to start each test from the same state we go to the home screen. Each test method then launches the app when required.